applications/GeoMechanicsApplication/custom_elements/small_strain_U_Pw_diff_order_element.cpp

```diff
@@ -669,27 +614,21 @@ void SmallStrainUPwDiffOrderElement::
 
     SizeType Index = 0;
 
-    if ( Dim > 2 ) {
-        for ( SizeType i = 0; i < NumUNodes; ++i ) {
-            rValues[Index++] = rGeom[i].FastGetSolutionStepValue( VELOCITY_X, Step );
-            rValues[Index++] = rGeom[i].FastGetSolutionStepValue( VELOCITY_Y, Step );
-            rValues[Index++] = rGeom[i].FastGetSolutionStepValue( VELOCITY_Z, Step );
-        }
-    }
-    else {
-        for ( SizeType i = 0; i < NumUNodes; ++i ) {
-            rValues[Index++] = rGeom[i].FastGetSolutionStepValue( VELOCITY_X, Step );
-            rValues[Index++] = rGeom[i].FastGetSolutionStepValue( VELOCITY_Y, Step );
+    for ( SizeType i = 0; i < NumUNodes; ++i ) {
+        rValues[Index] = rGeom[i].FastGetSolutionStepValue( VELOCITY_X, Step );
+        ++Index;
+        rValues[Index] = rGeom[i].FastGetSolutionStepValue( VELOCITY_Y, Step );
+        ++Index;
+        if ( Dim > 2 ) {
+            rValues[Index] = rGeom[i].FastGetSolutionStepValue( VELOCITY_Z, Step );
+            ++Index;
```

Comment:
If I'm correct, this was a part which wasn't tested. Do we still want to make these changes then?